lib/package_utils.rb

```diff
@@ -30,4 +31,23 @@ def self.get_sha256(pkg, build_from_source: false)
       return pkg.source_sha256
     end
   end
+
+  def self.get_binary_url(pkg)
+    # List all the packages with the name and version of the package file.
+    # The name search is fuzzy, so we need to refine it further (otherwise packages like vim, gvim and vim_runtime would break).
+    packages = JSON.parse(Net::HTTP.get(URI("https://gitlab.com/api/v4/projects/26210301/packages?package_name=#{pkg.name}&package_version=#{pkg.version}_#{ARCH}")))
+    # Loop over each result until we get an exact name match, then return the package ID for that match.
+    package_id = 0
+    (0..packages.count - 1).each do |i|
+      next unless packages[i]['name'] == pkg.name
+      package_id = packages[i]['id']
+    end
+    # List all the package files for that package ID.
+    package_files = JSON.parse(Net::HTTP.get(URI("https://gitlab.com/api/v4/projects/26210301/packages/#{package_id}/package_files")))
```

Comment:
Can `https://gitlab.com/api/v4/projects/26210301/packages` be a variable since it is used more than once?  The reason I say this is because the url may change in the future and it would be good to only change on one line.

Reply:
If the URL changes to anything other than gitlab, this whole thing needs to be rewritten-- I don't think a variable makes sense just to store `26210301` twice.